src/embeddings/embeddings_calculator.cc

```diff
@@ -88,62 +89,42 @@ class EmbeddingsCalculator : public CalculatorBase {
-                    for (auto& input : it->value.GetArray()) {
-                        if (!input.IsString())
-                            return absl::InvalidArgumentError("every element in input array should be string");
-                        input_strings.push_back(input.GetString());
-                    }
-                } else {
-                    return absl::InvalidArgumentError("input should be string or array of strings");
-                }
-            } else {
-                return absl::InvalidArgumentError("input field is required");
-            }
-            it = payload.parsedJson->FindMember("encoding_format");
-            if (it != payload.parsedJson->MemberEnd()) {
-                if (it->value.IsString()) {
-                    if (it->value.GetString() == std::string("base64")) {
-                        isBase64 = true;
-                    }
-                } else {
-                    return absl::InvalidArgumentError("encoding_format should be string");
-                }
-            }
-            // TODO: dimensions (optional)
-            // TODO: user (optional)
-        } else {
+        if (cc->Inputs().Tag(INPUT_TAG_NAME).IsEmpty()) {
             return absl::InvalidArgumentError("Input is empty");
         }
+        InputDataType payload = cc->Inputs().Tag(INPUT_TAG_NAME).Get<InputDataType>();
+        SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Request body: {}", payload.body);
+        SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Request uri: {}", payload.uri);
+        auto request = EmbeddingsRequest::from_json(payload.parsedJson);
+        if (auto error = std::get_if<std::string>(&request)) {
+            return absl::InvalidArgumentError(*error);
+        }
+        EmbeddingsRequest embeddingsRequest = std::get<EmbeddingsRequest>(request);
 
         // Automatically deduce tokenizer input name
         std::vector<std::string> tokenizerInputNames = tokenizer_session->getInputNames();
         std::vector<std::string> embeddingsInputNames = embeddings_session->getInputNames();
+
         RET_CHECK(tokenizerInputNames.size() == 1);
         const std::string& tokenizerInputName = tokenizerInputNames.at(0);
         SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Tokenizer input name detected: {}", tokenizerInputName);
 
         ::InferenceInput tokenizerInputMap;
-        tokenizerInputMap[tokenizerInputName] = ov::Tensor{
-            ov::element::string,
-            ov::Shape{input_strings.size()},
-            input_strings.data()};
 
         ::InferenceOutput embeddingsOutputMap;
+        size_t expected_batch_size = 1;
         try {
-            ::InferenceOutput tokenizerOutputMap = tokenizer_session->infer(tokenizerInputMap);
+            ::InferenceOutput tokenizerOutputMap;
+            if (auto strings = std::get_if<std::vector<std::string>>(&embeddingsRequest.input)) {
+                expected_batch_size = strings->size();
```

Comment:
Why do we need this variable? And why setting it only in this if statement? I think it was more readable previously

Reply:
It is meant as a futureproofing for when the expected batch size will be dependent on either the amount of input strings or the amount of int vectors.